src/st/strategy/shortcode/class-wpml-pb-update-shortcodes-in-content.php

```diff
@@ -80,17 +80,25 @@ private function update_shortcode_attributes( $shortcode_data ) {
 
 	private function replace_string_with_translation( $block, $original, $translation, $is_attribute = false, $attr = '' ) {
 		$translation = apply_filters( 'wpml_pb_before_replace_string_with_translation', $translation, $is_attribute );
+		$new_block   = $block;
 
-		$new_block = $block;
 		if ( $translation ) {
-			if ( $is_attribute && $attr ) {
-				$pattern   = '/' . $attr . '=(["\'])' . preg_quote( $original, '/' ) . '(["\'])/';
-				$new_block = preg_replace( $pattern, $attr . '=${1}' . $translation . '${2}', $block );
+			$is_text_too_long_for_regex = mb_strlen( $original ) > 5000;
```

Comment:
@strategio I would move this to a private method and the `5000` to a constant.

Reply:
Done @andreasciamanna